package/MDAnalysis/analysis/gnm.py

```diff
@@ -306,49 +313,29 @@ def generate_kirchoff(self):
 
         return matrix
 
-    def run(self, start=None, stop=None, step=None):
-        """Analyze trajectory and produce timeseries.
-
-        Parameters
-        ----------
-        start : int (optional)
-        stop : int (optional)
-        step : int (optional)
-
-        Returns
-        -------
-        results : list
-            GNM results per frame::
-
-                results = [(time,eigenvalues[1],eigenvectors[1]),(time,eigenvalues[1],eigenvectors[1])... ]
-
-        .. versionchanged:: 0.16.0
-           use start, stop, step instead of skip
-        """
-        logger.info("GNM analysis: starting")
-
+    def _prepare(self):
         self.timeseries = []
-        self._timesteps = []
-
-        for ts in self.u.trajectory[start:stop:step]:
-            self._timesteps.append(ts.time)
-
-            matrix = self.generate_kirchoff()
-            try:
-                u, w, v = np.linalg.svd(matrix)
-            except np.linalg.LinAlgError:
-                print("\nFrame skip at", ts.time,
-                      "(SVD failed to converge). Cutoff", self.cutoff)
-                continue
-            #Save the results somewhere useful in some useful format. Usefully.
-            self._generate_output(
-                w,
-                v,
-                self.results,
-                ts.time,
-                matrix,
-                ReportVector=self.ReportVector,
-                counter=ts.frame)
+
+    def _single_frame(self):
+        matrix = self.generate_kirchoff()
+        try:
+            u, w, v = np.linalg.svd(matrix)
+        except np.linalg.LinAlgError:
+            print("\nFrame skip at", self._ts.time,
```

Comment:
How about a warning not a print statement here?
Also needs coverage.

Reply:
I changed the statement to print. SVD is quite robust and I haven't found a simple test case quickly. Maybe you guys have an idea?

Reply:
Is it possible to just patch the return of np.linalg.svd to trigger the exception on call?

If not, I'd be happy with a pragma no cover.

Reply:
Maybe there is a way. Let me check tomorrow.

Reply:
Probably something like this (this is probably not the best/prettiest way to do it)?

```python
import pytest
import numpy as np
from unittest.mock import patch


def run_svd():
    a = np.random.randn(9, 6) + 1j*np.random.randn(9,6)
    return np.linalg.svd(a, full_matrices=True)


def test_linalgexcept():
    with patch('numpy.linalg.svd', side_effect=np.linalg.LinAlgError('SVD error')):
        with pytest.raises(np.linalg.LinAlgError):
            run_svd()
```

Reply:
That looks nice. I took the case from `test_reload_offsets_if_offsets_readin_fails`